verification/src/solidity/compiler_fetcher.rs

```diff
@@ -104,15 +107,99 @@ impl TryFrom<(json::CompilerInfo, &Url)> for CompilerInfo {
     }
 }
 
+#[derive(Default)]
+struct RefreshableCompilerVersions(Arc<parking_lot::RwLock<CompilerVersions>>);
+
+impl From<CompilerVersions> for RefreshableCompilerVersions {
+    fn from(versions: CompilerVersions) -> Self {
+        Self(Arc::new(parking_lot::RwLock::new(versions)))
+    }
+}
+
+impl RefreshableCompilerVersions {
+    fn spawn_refresh_job(
+        &self,
+        versions_list_url: Url,
+        cron_schedule: &str,
+    ) -> Result<(), JobSchedulerError> {
+        log::info!("spawn version refresh job with schedule {}", cron_schedule);
+        let versions = Arc::clone(&self.0);
+
+        let job = Job::new_async(cron_schedule, move |_, _| {
+            let versions_list_url = versions_list_url.clone();
+            let versions = Arc::clone(&versions);
+
+            Box::pin(async move {
+                let refresh_result =
+                    RefreshableCompilerVersions::refresh_versions(versions, &versions_list_url)
+                        .await;
+                if let Err(err) = refresh_result {
+                    log::error!("error during version refresh: {}", err);
+                };
+            })
+        })?;
+        let scheduler = JobScheduler::new()?;
+        scheduler.add(job)?;
+        scheduler.start()
+    }
+
+    async fn refresh_versions(
+        versions: Arc<parking_lot::RwLock<CompilerVersions>>,
```

Comment:
Are you sure that using `parking_lot::RwLock` and not `tokio::sync::RwLock` cannot cause deadlocks?